### ISSUE TITLE
Secure Messaging - Migrate to Re-Engineered MHV Patient APIs

### ISSUE THREAD

**Issue by bischoffa** [department-of-veterans-affairs/va-mobile-app#5562]
# Project Status Overview

### Project Objective:

- MHV SM is making changes to their APIs that the Mobile app uses. In order for Mobile's SM functionality to continue Mobile must make changes to align with MHV SM 

 
### Current Dependencies / Blockers
<details>
<summary>Blockers</summary>

|Blocker | Team / Owner | Est to Resolution |
|-------|---------|-------| 
| Mobile dependent on MHV SM on making the changes to API so that Mobile can perform their work | MHV SM Christy Gales  | APIs started to be completed in Aug and will continue through November. Mobile to test when they become ready per coordination with MHV SM |

</details>

### Project Timeline
<details>
<summary>Milestones</summary>

|Step | Timing | 
|-------| ---------| 
| Mobile integration |  Completed July 2023   |
| Mobile testing |  Ongoing - Sept - Nov 2023 |
| MHV SM to release changes into product | Done between Sept - Nov   |

</details>

### Monthly Update 
<details>
<summary>Updates</summary>

|Month| Project Health | Progress & Key Accomplishments| Issues/Risks/Blockers|
|-------| ---------|--------|---------|
|July '23| Yellow | Mobile no action; MHV SM working on APIs | MHV SM APIs are not ready for Mobile to test | 
|Aug '23| Yellow | MHV SM APIs not yet ready for mobile to test with; met and MHV SM and confirmed upcoming timeline; BE mobile confirmed Mobile is integrated and only remaining work from mobile is to test when MHV SM is ready; BE mobile confirmation that Mobile will use new APIs automatically when MHV SM moves them into production | 
|Sep '23| Green | Some MHV APIs ready for Mobile to test, mobile completed some testing | Progress being made on other APIs |
|Oct '23| Green | Mobile will continue to test when MHV SM is ready. Expected for testing to end in October and MHV SM final production changes to be done in Nov | Some components for mobile to test have been delayed | 
| Nov'23| Green | Mobile only has 1 MHV SM API to test. Currently waiting on when it will be ready. Next date TBD. Aiming to hear back from Christy Gales week of 11/27| 
| Dec '23 | Green | Originally completed testing on 12/1 however informed on 12/5 there is additional testing requirements from VA that just came through that will require Mobile to test new API Header Keys. Told it should be ready for Mobile to test after the week of 12/11 | 

</details>

# Detailed Project Scope 

### Problem Statement
 -  MHV SM is making changes to their APIs that the Mobile app uses. In order for Mobile's SM functionality to continue Mobile must make changes to align with MHV SM 

### Assumptions and Risk 
 - Mobile is dependent on MHV SM APIs being completed

### Definition of Done 
<details>
<summary>Following needs to be true</summary>

|Item | Completed | 
|-------| ---------|
| MHV SM to update their APIs | Completed | 
| Mobile to integrate with new MHV APIs | Completed | 
| MHV SM API ready for pre-prod testing | Ongoing | 
| Mobile to assist with MHV SM API testing | Ongoing |
| MHV SM to release API changes to production | Ongoing |  

</details>

### Project Scope
<details>
<summary>Items in and out of scope</summary>

|In Scope | Out of Scope| 
|-------| ---------|
| MHV SM to update their APIs | | 
| Mobile to integrate with new MHV APIs | | 
| MHV SM API ready for pre-prod testing | | 
| Mobile to assist with MHV SM API testing | |
| MHV SM to release API changes to production | |  

</details>

### Important Project Links 
<details>
<summary>Links</summary>
- [Testing document 9/1/23](<img width="423" alt="image" src="https://github.com/department-of-veterans-affairs/va-mobile-app/assets/116006847/10f0c735-08bc-4595-8280-05331883e82b">)

</details>

### Team Members

<details>
<summary>Contact List</summary>

- **OCTO Product Leads:** Rachel Han
- **Ad Hoc PM:** Adam Kasmier
- **Engineering:** Jayson Perkins and Jon Bindbeutel
- **Design:** NA
- **MHV SM PO:** Christy Gales - PM  
- **External Team 1 & Role:** Satish Bhamidipaty - Dev 

</details>

### OKR and Analytic Metrics 
<details>
<summary>Metrics</summary>

Not applicable. Work specifically focused on ensuring SM continues to work within the mobile app. Mobile will coordinate with MHV SM and test for pre-production needs. Will monitor post launch if there are any issues. 

</details>

### Questions
<details>
<summary>Closed </summary>

- MHV SM will verify but are pretty sure there is no changes to responses in this change (our expectation is for it be the same as it is today) ; if there is then changes timing / effort - closed
- MHV found a defect in an SM API, need to verify if it impacts us - defect does not impact use as we do not use the API - Jayson confirmed - closed
- Will this impact Push Notifications - MHV SM said it would not - closed 
- MHV SM indicated they wanted to be careful on rollout due to capacity load - they indicated we did not have to worry about a slow rollout % - closed 

</details>

# Checklist

<details>
  <summary>Toggle Product Checklist</summary>

**Project Kickoff & Requirements Gathering**
   - [X] Project Kickoff session
   - [X] Understand MHV SM API impact 
   - [ ] 

**Pre-Production Readiness** - NA 

**Feature Go Live** - NA

</details>

<details>
  <summary>Toggle Team Collaboration Checklist - NA </summary>

</details>

<details>
  <summary>Toggle Design/UX Checklist - NA</summary>

- [ ] Review problem/opportunity statement
- [ ] Add questions and assumptions to product brief
- [ ] Add research links to product brief
- [ ] Document high-level research/design plans
- [ ] Create lo/hi fi wireframes & user flows if applicable
- [ ] Add links to Epic & applicable GitHub folders
- [ ] Socialize designs
- [ ] Sign off received
- [ ] Complete Design QA
- [ ] Does it require component review?
- [ ] Accessibility review needed?

</details>

<details>
  <summary>Toggle Content Checklist - NA</summary>

- [ ] Review product brief and surface any questions, assumptions & risks
- [ ] Participate in project discovery and kickoff activities/ceremonies
- [ ] Complete comparative analysis/content research activities for net-new content
- [ ] Review current content and determine where app may need to differ and potential improvements to share with web
- [ ] Review past VA research and decisions documentation
- [ ] Support Research & UX Design in research sessions and synthesis
- [ ] Make content recommendations for lo-/hi-fi wireframes in collaboration with UX Design
- [ ] Collaborate with Sitewide Content team for alignment, improvements, and sign-off
- [ ] Participate in FE hand off
- [ ] Complete content QA
- [ ] Write copy for What's New In App
- [ ] Write copy for app stores' What's New sections.
- [ ] Write copy for updated screenshots in app stores 

</details>

<details>
  <summary>Toggle Engineering Checklist</summary>

- [X] Are UI designs/ specs available/ready? If yes, they should be attached to  attach specs. (We are assuming the design specs have been approved)
- [X] Are there any deadlines that need to be met?
- [ ] What is the roll out plan? (MHV SM is controlling rollout, given its their APIs. Mobile will test work pre-production within the dates MHV SM provides)
- [X] Does this project require api integration?
- [X] Do we want to capture analytics for this projects (no, changes purely to make sure SM continues to function as expected in Mobile with the API changes. API maintenance work. 

</details>

<details>
  <summary>Toggle QA Checklist - NA </summary>

- [ ] Create test plan
  -  **Template: QA Test Plan**
- [ ] Identify test data needs
- [ ] Gotten & incorporated test plan review
- [ ] QA Complete
- [ ] Waygate tasks complete
- [ ] Added new feature cases to the RC in TestRail
- [ ] Added feature cases to appropriate section of active cases in TestRail
- [ ] Spun up a (future) UI automation ticket for new feature
</details>

**Comment by bischoffa** [comment]
8/22 Meeting 

Met with Christ / MHV SM Team 8/22

**Meeting notes:**
- MHV has there final 2 APIs in progress
- Expecting their original timeline of early Sept to be delayed - targeting finishing end of their PI increment 
- Identified that Mobile is already pointing to it 
   - Meaning once MHV is done Mobile will need to test - MHV SM to contact Mobile when ready 
   - Upon successful testing mobile / MHV to coordinate the switch over  
- Agreed that future meetings not needed unless something substantial comes up - instead will use email communication  
- Verified MHV Attachment changes are okay during the meeting - Mobile has seen no issue 

**Next Steps:** 

1.  Christy to share estimated times (noted that it could change) 
2.  MHV SM working to complete final 2 APIs
3.   MHV SM to email Mobile (same as current chain) when Mobile test window is 
4.   Mobile to test (provide feedback to MHV SM if needed) 
5.   MHV SM to determine switch date and communicate that to mobile
6.   MHV SM and Mobile to switch to newly engineered containers 

-

**Comment by bischoffa** [comment]
Received email today from MHV SM that testing is ready. Note that Mobile was not anticipating this as MHV SM follow-ups with updated timing was not shared. Working to see what we can do to support this request. 

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBblJiIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--4668982b6a6a6566102d0e4d0d9854860b10e051/image.png)

**Comment by bischoffa** [comment]
Estimating timing from Christ MHV SM 
Looks like Mobile will be doing some kind of testing each sprint in Sept - Nov based on the list below. 

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBdEJnIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--562f54c9a3d668b6b9bf9cc0c6e1404b6664d963/image.png)

**Comment by bischoffa** [comment]
Removed Q3 2023 label as work was not done in Q3 as originally planned, moved to Q4 given MHV Web has had some delays on their end that has shifted to Mobile targeting completing this work in Q4 2023

**Comment by bischoffa** [comment]
Updates made to tickets and not on epic so updated now for past history. Many of these updates come from DMs between Christy Gales and myself. 

*   10/30 Phase 2 send and reply work MHV was doing was rolled back and wont be available for mobile to test till around 11/15 
*   11/3 Mobile followed up and informed that Send/Reply back in SySTest and ready to be tested ; Phase 3 work will not be ready for mobile till either 11/22 or 11/29 given holidays 
*   11/13 told Phase 3 needs to go out in production on 11/20 but its not yet in testing environment to be tested 
*   11/16 Mobile reached out for an update given production was 2 days away and informed they are still working on a few defects related to a configuration problem and not code 
*   11/17 Phase 3 cannot be tested will 11/19 so mobile plans for it accordingly
*   11/30 Mobile asks for Phase 3 update - MHV informs ongoing configuration issue that they thought was fixed - no eta 
*   12/1 Christy tells Mobile Phase 3 is ready for testing 
*   12/5 Mobile successfully tests Phase 3 ; Mobile informed by MHV SM that unplanned new testing is needed on API Header Keys - told we just need to test send message / reply w attachment to make sure it works as expected
*   12/11 Mobile [informed](https://dsva.slack.com/archives/C05TU9FPZU0/p1702333210816559) Phase 3 testing is needed again and API Header Key Testing is ready
*   12/12 Mobile completes testing; verifies with Christy that no other planned work is expected for this project

Mobile to close out this project. If new work comes up Christy informed to chat with Janie who will work with Rachel (Mobile PO) on future work.
